File: src/solutions/c001/solution.go
package main

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for index, value := range nums{
		num := target - value
		if _, ok := m[num]; ok {
			return []int{m[num], index}
		}
		m[value] = index
	}
	return nil
}

File: src/solutions/c001/solution.py
from typing import List

from builtins import int


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = dict()
        index = 0
        for num in nums:
            idx = m.get(target - num)
            if idx is not None:
                return [idx, index]
            m[num] = index
            index += 1
        return None
